kvdb-web/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A key-value database for use in browsers
+//!
+//! An example on how to use it in a browser can be seen kvdb_web_demo crate.
+
+#![cfg(target_arch = "wasm32")]
+
+mod hex;
+
+pub use kvdb::{DBValue, DBTransaction, DBOp, KeyValueDB};
+pub use wasm_bindgen::{JsCast, UnwrapThrowExt};
+pub use web_sys::console;
+
+use wasm_bindgen::prelude::*;
+use web_sys::Storage;
+
+use std::collections::HashMap;
+use std::io;
+use std::sync::{Arc, RwLock};
+
+use serde::{Serialize, Deserialize};
+use crate::hex::BytesHexEncoding;
+
+
+type Key = BytesHexEncoding;
+type Value = BytesHexEncoding;
+
+macro_rules! console_log {
```

Comment:
We should just use `log!` and `warn!` from the `log` crate, and have them redirect to the console, for example using https://crates.io/crates/console_log